pegasus/sites.v3/code.org/views/census_map.haml

```diff
@@ -81,9 +104,56 @@
 
     google.maps.event.addListener(layer, 'click', function(event) {
       enableDrag();
-      event.infoWindowHtml = event.row['organization_name'].value;
-      event.infoWindowHtml += ' (' + event.row['city'].value + ')';
-      event.infoWindowHtml += '<br>';
+      let censusMessage;
+      switch(event.row['teaches_cs'].value) {
+        case 'YES':
+          censusMessage = "We believe this school offers Computer Science class(es).";
+          break;
+        case 'NO':
+          censusMessage = "We believe this school offers limited or no Computer Science opportunities.";
+          break;
+        case 'MAYBE':
+          censusMessage = "We are unsure if this school offers Computer Science class(es).";
+          break;
+        default:
+          censusMessage = "We have no data about this school.";
+      }
+      const schoolDropdownOptionString = `{
+        value: '${event.row['school_id'].value}',
+        label: '${event.row['school_name'].value}',
+        school: {
+          nces_id: '${event.row['school_id'].value}',
+          name: '${event.row['school_name'].value}',
+          city: '${event.row['city'].value}',
+          state: '${event.row['state_code'].value}',
+          latitude: ${event.row['location'].value.split(',')[0]},
+          longitude: ${event.row['location'].value.split(',')[1]},
+        },
+      }`;
+      event.infoWindowHtml  = '<div class="census-info-window">';
```

Comment:
I haven't seen this in our codebase before, but that doesn't mean it's not the right way to do it.  Could you make this a React component that is hidden until you pass it the info it needs and then show it?

Reply:
This is how the map works currently. I am just changing the contents of the info window. The fusion table layer seems to work by displaying the contents of `event.infoWindowHtml` after this event is processed. I don't know of a way to be notified when that HTML is inserted into the DOM so it would be hard to attach a React component at the right time.